inc/console/task/unlockcommand.class.php

```diff
@@ -0,0 +1,161 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Console\Task;
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+use CronTask;
+use Glpi\Console\AbstractCommand;
+use Glpi\Event;
+use QueryExpression;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+
+class UnlockCommand extends AbstractCommand {
+
+   protected function configure() {
+      parent::configure();
+
+      $this->setName('glpi:task:unlock');
+      $this->setAliases(['task:unlock']);
+      $this->setDescription(__('Unlock automatic tasks'));
+
+      $this->addOption(
+         'all',
+         'a',
+         InputOption::VALUE_NONE,
+         __('Unlock all tasks')
+      );
+
+      $this->addOption(
+         'cycle',
+         'c',
+         InputOption::VALUE_OPTIONAL,
+         __('Number of run cycles from which the task is considered as stuck (delay = frequency * cycle)'),
+         25
+      );
+
+      $this->addOption(
+         'task',
+         't',
+         InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+         __('Name of task to unlock')
+      );
+   }
+
+   protected function execute(InputInterface $input, OutputInterface $output) {
+
+      $all   = $input->getOption('all');
+      $cycle = $input->getOption('cycle');
+      $tasks = $input->getOption('task');
+
+      if (!preg_match('/^\d+$/', $cycle)) {
+         throw new InvalidArgumentException(
+            __('Option --cycle has to be an integer.')
+         );
+      }
+
+      if ($all && !empty($tasks)) {
+         throw new InvalidArgumentException(
+            __('Option --all is not compatible with option --task.')
+         );
+      }
+
+      if (!$all && empty($tasks)) {
+         throw new InvalidArgumentException(
+            __('You have to specify which asks to unlock using --all or --task options.')
+         );
+      }
+
+      $task_iterator = $this->db->request(
+         [
+            'SELECT' => [
+               'id',
+               'name',
+            ],
+            'FROM'   => Crontask::getTable(),
+            'WHERE'  => [
+               'state' => Crontask::STATE_RUNNING,
+               new QueryExpression(
+                  'UNIX_TIMESTAMP(' .  $this->db->quoteName('lastrun') . ')'
+                  . ' + ' . $cycle
+                  . ' * ' . $this->db->quoteName('frequency')
+                  . ' <  UNIX_TIMESTAMP(NOW())'
+               )
+            ]
+         ]
+      );
+
+      $crontask = new CronTask();
+      $unlocked_count = 0;
+
+      foreach ($task_iterator as $task) {
+         if (!$all && !in_array($task['name'], $tasks)) {
```

Comment:
Potentially, the name could ambiguous. I think we should change this argument to use a format like `itemtype.name` that cannot be ambiguous as the unicity key is buld on these columns.

Reply:
Done.